### PR TITLE
Psalm level 6 5

### PR DESCRIPTION
**Pull request by carlos-granados**
Applies all the remaining rules needed to reach Psalm level 6

### CODE REVIEW

File: src/Behat/Testwork/Argument/MixedArgumentOrganiser.php [hunk 1]
```diff
@@ -241,13 +243,10 @@ private function getReflectionClassesFromParameter(\ReflectionParameter $paramet
 
         $type = $parameter->getType();
 
-        /**
```
Comment (carlos-granados) [inline src/Behat/Testwork/Argument/MixedArgumentOrganiser.php:244]: This is not needed not that we are in PHP 8

File: src/Behat/Testwork/Autoloader/ServiceContainer/AutoloaderExtension.php [hunk 2]
```diff
@@ -78,7 +78,9 @@ public function configure(ArrayNodeDefinition $builder)
             ->treatTrueLike($this->defaultPaths)
             ->treatNullLike(array())
             ->treatFalseLike(array())
-
+            ;
```
Comment (carlos-granados) [inline src/Behat/Testwork/Autoloader/ServiceContainer/AutoloaderExtension.php:81]: The return value of `treatFalseLike()` is a NodeDefinition, which does not have the `prototype()` function, so we need to narrow down the actual type

File: src/Behat/Testwork/Ordering/Cli/OrderController.php [hunk 3]
```diff
@@ -77,14 +77,15 @@ public function execute(InputInterface $input, OutputInterface $output)
         $orderer = $input->getOption('order');
 
         if (!$orderer) {
-            return;
+            return null;
```
Comment (carlos-granados) [inline src/Behat/Testwork/Ordering/Cli/OrderController.php:80]: This function needs to return a value

File: src/Behat/Testwork/Output/Printer/JUnitOutputPrinter.php [hunk 4]
```diff
@@ -141,8 +141,9 @@ public function setFileName($fileName, $extension = 'xml')
         if ('.'.$extension !== substr($fileName, strlen($extension) + 1)) {
             $fileName .= '.'.$extension;
         }
-
-        $this->getOutputFactory()->setFileName($fileName);
+        $outputFactory = $this->getOutputFactory();
```
Comment (carlos-granados) [inline src/Behat/Testwork/Output/Printer/JUnitOutputPrinter.php:144]: Again need to narrow down the actual type

**Review by acoulton — APPROVED** [review]
Great :)

**Pull request merged**